### PR TITLE
[slack] populate users on member channel join

### PR DESCRIPTION
**Pull request by dajohi**

### CODE REVIEW

**Comment by patcon** [comment]
Thanks @dajohi! Great find.

I don't know this section of codebase so well, but wondering if maybe we could just insert the user in question from the payload so that it's less intensive, like here:

https://github.com/42wim/matterbridge/blob/f5659d455d2c28a6f2fe4c40f4dc344419ff523a/bridge/slack/handlers.go#L80-L83

**Comment by dajohi** [comment]
@patcon request for your second review :)

File: bridge/slack/helpers.go [hunk 1]
```diff
@@ -63,6 +63,18 @@ func (b *Bslack) getChannelBy(lookupKey string, lookupMap map[string]*slack.Chan
 
 const minimumRefreshInterval = 10 * time.Second
 
+func (b *Bslack) populateUser(userID string) {
+	user, err := b.sc.GetUserInfo(userID)
+	if err != nil {
```
Comment (Helcaraxan) [inline bridge/slack/helpers.go]: Only thing that I would add here is handling rate-limiting correctly.

You can see the typical way of doing this in the code of the `populateUsers()` method below.

**Comment by aslafy-z** [comment]
Any chance you finish this (quite critical) PR @dajohi? Thanks!

File: bridge/slack/helpers.go (hunk 1 above)
Comment (dajohi) [reply bridge/slack/helpers.go]: I decided to check the cache instead of implementing rate-limiting code.  This is much simpler.

**Comment by 42wim** [comment]
LGTM, thanks for your work @dajohi !

**Pull request merged**